src/main/java/fr/openent/competences/helpers/NoteControllerHelper.java

```diff
@@ -0,0 +1,32 @@
+package fr.openent.competences.helpers;
+
+import fr.openent.competences.constants.Field;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+
+public class NoteControllerHelper {
+
+    protected static final Logger log = LoggerFactory.getLogger(DevoirControllerHelper.class);
+
+    public static JsonObject setResponseExportReleve(JsonObject responseAnnual, JsonObject responsePeriodic) {
+
+        JsonArray annualStudents = responseAnnual.getJsonArray(Field.ELEVES);
+        JsonArray periodicStudents = responsePeriodic.getJsonArray(Field.ELEVES);
+
+        for(int i = 0; i < periodicStudents.size(); i++){
+            JsonObject studentPeriodicJO = periodicStudents.getJsonObject(i);
+            for (int j = 0; j < annualStudents.size(); j++) {
+                JsonObject studentAnnualJO = annualStudents.getJsonObject(j);
+                if (studentPeriodicJO.getString(Field.ID).equals(studentAnnualJO.getString(Field.ID))) {
+                    studentPeriodicJO.put(Field.AVERAGES, studentAnnualJO.getJsonArray(Field.MOYENNES, new JsonArray()));
+                    studentPeriodicJO.put(Field.FINALAVERAGES, studentAnnualJO.getJsonArray(Field.MOYENNESFINALES, new JsonArray()));
+                }
+            }
+        }
```

Comment:
Version stream
```suggestion
        // This function allows to find an annualStudent from a periodicStudent
        Function<JsonObject, JsonObject> getAnnualStudent = periodicStudent -> annualStudents.stream()
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
                .filter(studentAnnualJO -> periodicStudent.getString(Field.ID).equals(studentAnnualJO.getString(Field.ID)))
                .findFirst()
                // If we don't find anything we return an empty jsonObject
                .orElse(new JsonObject());

        periodicStudents.stream()
                .filter(JsonObject.class::isInstance)
                .map(JsonObject.class::cast)
                .collect(Collectors.toMap(Function.identity(), getAnnualStudent))
                // From here we have a map of periodicStudent in key and an annualStudent in value
                .entrySet().stream()
                // We may not have found an annualStudent, so we filter them
                .filter(periodicAnnualStudentsEntry -> !periodicAnnualStudentsEntry.getValue().isEmpty())
                .forEach(periodicAnnualStudentsEntry -> {
                    periodicAnnualStudentsEntry.getKey().put(Field.AVERAGES, periodicAnnualStudentsEntry.getValue().getJsonArray(Field.MOYENNES, new JsonArray()));
                    periodicAnnualStudentsEntry.getKey().put(Field.FINALAVERAGES, periodicAnnualStudentsEntry.getValue().getJsonArray(Field.MOYENNESFINALES, new JsonArray()));
                });
```

Reply:
Sinon il faut arrêter la boucle des que l'on a trouvé un annualStudents qui correspond a notre condition
```suggestion
        for(int i = 0; i < periodicStudents.size(); i++){
            JsonObject studentPeriodicJO = periodicStudents.getJsonObject(i);
            for (int j = 0; j < annualStudents.size(); j++) {
                JsonObject studentAnnualJO = annualStudents.getJsonObject(j);
                if (studentPeriodicJO.getString(Field.ID).equals(studentAnnualJO.getString(Field.ID))) {
                    studentPeriodicJO.put(Field.AVERAGES, studentAnnualJO.getJsonArray(Field.MOYENNES, new JsonArray()));
                    studentPeriodicJO.put(Field.FINALAVERAGES, studentAnnualJO.getJsonArray(Field.MOYENNESFINALES, new JsonArray()));
                    break;
                }
            }
        }
```

Reply:
Pour le moment, on ne va pas chercher à faire de l'optimisation de code (trop compliqué pour actuellement).
On va d'abord faire une initiation au stream en douceur.
Donc optimisation de lisibilité de code plus qu'optimisation de code !